miner/src/worker/mod.rs

```diff
@@ -103,5 +118,11 @@ pub fn start_worker(
 }
 
 pub trait Worker {
-    fn run(&mut self, progress_bar: ProgressBar);
+    fn run(&mut self, nonce_range: Range<u64>, progress_bar: ProgressBar);
```

Comment:
Alternative, we can just pass a random generator into it.

```suggestion
    fn run<G: FnMut() -> u64>(&mut self, rng: G, progress_bar: ProgressBar);
```